pkg/framework/aries/api/factory.go

```diff
@@ -8,9 +8,15 @@ package api
 
 import (
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/didresolver"
 )
 
 // TransportProviderFactory allows overriding of aries protocol providers
 type TransportProviderFactory interface {
 	CreateOutboundTransport() transport.OutboundTransport
 }
+
+// DIDResolverProviderFactory allows overriding of aries did resolver providers
+type DIDResolverProviderFactory interface {
+	CreateDIDResolverOpt() didresolver.Opt
```

Comment:
As a general comment, I think this is intended to create a DID Resolver. I'm unclear what the meaning of "Opt" is.

Reply:
Updated to configure DIDResolver rather than lower level configs.